frontend/src/component/feature/FeatureStrategy/NewFeatureStrategyCreate/NewFeatureStrategyCreate.test.tsx

```diff
@@ -199,20 +199,45 @@ describe('NewFeatureStrategyCreate', () => {
         const addVariantEl = await screen.findByText('Add variant');
         fireEvent.click(addVariantEl);
 
+        const inputElement = screen.getAllByRole('textbox')[0];
+        fireEvent.change(inputElement, {
+            target: { value: expectedVariantName },
+        });
+
         const targetingEl = await screen.findByText('Targeting');
         fireEvent.click(targetingEl);
 
         const addConstraintEl = await screen.findByText('Add constraint');
         expect(addConstraintEl).toBeInTheDocument();
 
         fireEvent.click(variantsEl);
+        const inputElement2 = screen.getAllByRole('textbox')[0];
 
-        const inputElement = screen.getAllByRole('textbox')[0];
-        fireEvent.change(inputElement, {
-            target: { value: expectedVariantName },
-        });
+        expect(inputElement2).not.toBeDisabled();
+    });
 
-        expect(screen.getByText(expectedVariantName)).toBeInTheDocument();
+    test('should remove empty variants when changing tabs', async () => {
+        setupComponent();
+
+        const titleEl = await screen.findByText('Gradual rollout');
+        expect(titleEl).toBeInTheDocument();
+
+        const variantsEl = screen.getByText('Variants');
+        fireEvent.click(variantsEl);
+
+        const addVariantEl = await screen.findByText('Add variant');
+        fireEvent.click(addVariantEl);
+
+        const targetingEl = await screen.findByText('Targeting');
+        fireEvent.click(targetingEl);
+
+        const addConstraintEl = await screen.findByText('Add constraint');
+        expect(addConstraintEl).toBeInTheDocument();
+
+        fireEvent.click(variantsEl);
+
+        const variants = screen.queryAllByTestId('VARIANT');
+        expect(variants.length).toBe(0);
```

Comment:
could we add a check before this one that shows that the length was non 0 before?